scripts/animatediff_output.py

```diff
@@ -117,6 +121,50 @@ def _interp(
 
         return video_list
 
+    def _returnAnimatedImage(
+        self,
+        params: AnimateDiffProcess,
+        image_list: list,
+        res: Processed,
+        index: int,
+    ):
+        pil_image_list = []
+        image_array = [np.array(v) for v in image_list]
+        infotext = res.infotexts[index]
+        use_infotext = shared.opts.enable_pnginfo and infotext is not None
+        from io import BytesIO
+        import imageio.v2 as imageioV2
+        if "GIF" in params.format:
+            buffer = BytesIO()
+            imageioV2.mimwrite(
+                buffer, 
+                image_array, 
+                format='GIF', 
+                duration=(1000 / params.fps), 
+                loop=params.loop_number,
+                comment=(infotext if use_infotext else "")
+            )
+            pil_image_list.append(Image.open(buffer))
+        if "WEBP" in params.format:
+            buffer = BytesIO()
+            if PIL.features.check('webp_anim'):          
+                exif_bytes = b''
+                if use_infotext:
+                    exif_bytes = piexif.dump({
+                        "Exif":{
+                            piexif.ExifIFD.UserComment:piexif.helper.UserComment.dump(infotext, encoding="unicode")
+                        }})
+                lossless = shared.opts.data.get("animatediff_webp_lossless", False)
+                quality = shared.opts.data.get("animatediff_webp_quality", 80)
+                imageioV2.mimwrite(buffer, image_array, format='WEBP', plugin='pillow',
+                    duration=int(1 / params.fps * 1000), loop=params.loop_number,
+                    lossless=lossless, quality=quality, exif=exif_bytes
+                )
+                pil_image_list.append(Image.open(buffer))
+            else:
+                logger.warn("WebP animation in Pillow requires system WebP library v0.5.0 or later")
+        return pil_image_list
```

Comment:
This method has too much code duplication with the method below. Also, I'm not aware of any way that PIL can work with MP4. I would prefer to patch [encode_pil_to_base64](https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/api/api.py#L104) in WebUI in this way - if the output is already bytes, let's return directly; otherwise, encode it